example/example8.cpp

```diff
@@ -70,8 +70,8 @@ class MyObject3 : public std::enable_shared_from_this<MyObject3> {
 };
 
 /// Make pybind aware of the ref-counted wrapper type (s)
-PYBIND11_DECLARE_HOLDER_TYPE(T, ref<T>);
```

Comment:
There is no `;` at the end of this macro, it injects namespace into the code, and hence the warning `;` after namespace closing brace.

But I'm not too concerned about warnings in the example code. User of the library can choose how to use this macro. And I agree that macros should look like proper function calls and this sort of missing semicolon style is ugly. But the typical work around of wrapping macro in do{}while(0) doesn't work in this case because of namespace.